src/module/RegisterMappingFactory.cpp

```diff
@@ -31,54 +31,176 @@ std::shared_ptr<RegisterMapping> RegisterMappingFactory::fromJSONMapping(const M
     switch (jsonMapping.getDataType())
     {
     case RegisterMapping::OutputType::BOOL:
+    {
+        // Make place for the mapping that should be returned
+        auto mapping = std::shared_ptr<BoolMapping>{};
+
+        // Check if the default value is anything
+        auto lowerCase = jsonMapping.getDefaultValue();
+        std::transform(lowerCase.cbegin(), lowerCase.cend(), lowerCase.begin(), ::tolower);
+        bool* defaultValue = nullptr;
+        if (lowerCase == "true")
+            defaultValue = new bool(true);
```

Comment:
Don't do new never, except inside smart pointer constructor.
Mapping takes a pointer since the value is optional, you may just pass the address of local variable instead of creating one on the heap